include/seqan/basic/iterator_interface.h

```diff
@@ -350,43 +350,12 @@ position(TIterator const & it,
 
 template <typename T, typename TContainer>
 inline SEQAN_HOST_DEVICE bool
-atBegin(T const & it, TContainer const & cont)
-{
-    SEQAN_CHECKPOINT;
-    return it == begin(cont, Standard());
-}
-
-template <typename T, typename TContainer>
-inline SEQAN_HOST_DEVICE bool
-atBegin(T const & it, TContainer & cont)
-{
-    SEQAN_CHECKPOINT;
-    return it == begin(cont, Standard());
-}
-
-template <typename T, typename TContainer>
-inline SEQAN_HOST_DEVICE bool
 atBegin(T & it, TContainer & cont)
```

Comment:
Well, this could happen with the non-const interface as well. 
Secondly, this shouldn't cause any problems as the comparison of two iterators should be independent of there const-ness, as it is checked if they point to the same element requiring them to be in the same domain. Since the comparison does not change the state it should be const. 
I mean what is the benefit of the non-const version?
Moreover, something like this could not work:
```cpp
if (atEnd(std::find(begin(cont), end(cont), f), cont))
{
    // do stuff
}
```
Since the rvalue cannot bind to the lvalue reference, but to a lvalue const reference.
I know the example is not very intuitive, but since iterators are considered lightweight objects they might as well be returned by value in some functions. So the const version does not harm and is safer in usage.
The same for the other case. Assume, that the container of the iterator is lightweight and I call a function to get the container by value, the non-const version wouldn't work.
Or am I overlooking something here?
As far as I know, the comparison is implemented on const references anyway. So why shouldn't that work?